cpp/dolfinx/fem/DirichletBC.h

```diff
@@ -495,76 +508,82 @@ class DirichletBC
   void set(std::span<T> x, std::optional<std::span<const T>> x0,
            T alpha = 1) const
   {
-    std::int32_t x_size = x.size();
-    if (alpha == T(0)) // Optimisation for when alpha == 0
+    // set_fn is a lambda which gets evaluated for every index in [0,
+    // _dofs0.size()) and its result is assigned to x[_dofs0[i]].
+    auto apply = [&](std::invocable<std::int32_t> auto set_fn)
```

Comment:
Is it possible to avoid capturing everything in the lambda function?

Reply:
It is changed to it, it is not very neat however - we need to be quite verbose to make sure we do not capture a copy or a reference when we don't want it. Revert?